tests/test_applications.py

```diff
@@ -534,3 +534,14 @@ def get_app() -> ASGIApp:
     test_client_factory(app).get("/foo")
 
     assert SimpleInitializableMiddleware.counter == 2
+
+
+def test_lifespan_typing():
+    class App(Starlette):
+        pass
+
+    @asynccontextmanager
+    async def lifespan(app: App) -> AsyncIterator[None]:  # pragma: no cover
+        yield
+
+    App(lifespan=lifespan)
```

Comment:
Does this test fails before this PR?

Reply:
No, not really, I’m not sure how to force it to fail since we don’t run mypy on tests right?

Reply:
Maybe assert_type on the constructor? Idk

Reply:
Ok mypy does pick it up:

Before this change:
```
tests/test_applications.py:550: error: Argument "lifespan" to "App" has incompatible type "Callable[[App], _AsyncGeneratorContextManager[None]]"; expected "Union[Callable[[Starlette], AbstractAsyncContextManager[None]], Callable[[Starlette], AbstractAsyncContextManager[Mapping[str, Any]]], None]"  [arg-type]
Found 1 error in 1 file (checked 1 source file)
```